### PR TITLE
Fix a false positive for `Style/CaseLikeIf` when conditional contains comparison with a class

### LINKED ISSUE #8508: Bad autocorrect for Style/CaseLikeIf
In rubocop 0.89.0

If you have code like this:

```ruby
if klass == ClassA
  do_something
elsif klass == ClassB
  do_something_else
end
```

It incorrectly gets autocorrected to:

```ruby
 case klass
 when ClassA
   do_something
 when ClassB
   do_something_else
 end
```

@marcandre warned about this here https://github.com/rubocop-hq/rubocop/issues/7736#issuecomment-634915216 but I think this advice went unheeded.

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #8508

### CODE REVIEW

File: lib/rubocop/cop/style/case_like_if.rb [hunk 1]
```diff
@@ -202,6 +214,16 @@ def const_reference?(node)
             name == name.upcase
         end
 
+        def class_reference?(node)
+          return false unless node.const_type?
+
+          name = node.children[1].to_s
```
Comment (marcandre) [inline lib/rubocop/cop/style/case_like_if.rb:220]: How about using `node.children[1].match? /[[:lower:]]/` instead?

Comment (fatkodima) [reply lib/rubocop/cop/style/case_like_if.rb:220]: The original code checks that const name is not all uppercase, so the code could be written as `!name.match?(/^[[:upper:]]+$/)`, but this does not handle numbers in names and so, the original code looks simpler to me.

**Comment by fatkodima** [comment]
Marked it as unsafe.

File: lib/rubocop/cop/style/case_like_if.rb (hunk 1 above)
Comment (marcandre) [reply lib/rubocop/cop/style/case_like_if.rb:220]: > checks that const name is not all uppercase

Right. I'm sorry though, I don't see how "not all uppercase" != "has a lowercase", could you provide an example?

Comment (fatkodima) [reply lib/rubocop/cop/style/case_like_if.rb:220]: You're right, sorry. I mistakenly matched that regexp only with the first letter 😄

Comment (fatkodima) [reply lib/rubocop/cop/style/case_like_if.rb]: Updated.

**Comment by marcandre** [comment]
I'll merge after the merge conflict is resolved... 😢

**Comment by marcandre** [comment]
Thanks!

**Pull request merged**